controllers/trainingjob/trainingjob_controller.go

```diff
@@ -298,6 +321,19 @@ func (r *Reconciler) removeFinalizer(ctx reconcileRequestContext) error {
 	return nil
 }
 
+// Add information regarding the debugging statuses to the status fields.
+func (r *Reconciler) addDebugRuleEvaluationStatusesToStatus(ctx reconcileRequestContext) error {
```

Comment:
This function is named the same way as `addModelPathToStatus` but does something completely different -- it does not call `r.Status().Update()`. We need to be consistent or be excruciatingly clear about the difference.. I don't see any reason for this function to not call `r.Status.Update()`.

Reply:
Agree.